### ISSUE TITLE
Ambiguous empty choices

### ISSUE THREAD

**Issue by codalogic** [arineng/jcr#31]
tl;dr: After some analysis I don't think this is a concern, but it might be worth documenting it in an appendix or similar as others may ask the same question.

----------------------

When a choice or array has zero or one member, there's no `,` or `|` character to say whether it is a sequence or choice. This is primarily an issue when a type is expected to be extended in a later version.

For example, are these likely to be sequences or choices in future versions?:

    []
    {}
    [ "MountDrive" ]
    { "MountDrive" : string }

If it's an array that is expected to be extended, there may not be an issue.  The initial version of an array containing boot phases might be:

    $boot_phases = [ string * ]

A later version could become:

    $boot_phases = [ ("MountDrive" | string) * ]

For an object that is supposed to be a choice, we could initially do:
```
$shutdown_reason = { // : any ? }
```

when another option is defined, this would be changed to:
```
$shutdown_reason = { "MountDrive" : string | // : any ? }
```

So it looks like that initial concern is not an issue in reality. But since others may ask the same question it would be worth documenting.

**Comment by codalogic** [comment]
This becomes more problematic when we adopt something like the @{augments} annotation.

e.g. if you have:
```
$extension_point = { }
...
$extension1 = @{augments $extension_point} "foo" : string
$extension2 = @{augments $extension_point} "bar" : string
```
should `$extension_point` become:
```
$extension_point = { $extension1, $extension2 }
```
or:
```
$extension_point = { $extension1 | $extension2 }
```

It would be nice to be able to assume `$extension_point` is a sequence unless it is indicated as a choice, either by doing:
```
$extension_point = { | }
```
```
$extension_point = { "initial" : int | }
```
```
$extension_point = @{choice} { }
```
```
$extension_point = @{choice} { "initial" : int }
```
The `@{choice}` annotation looks like the easiest option from where we are.
